setuptools-scm seems to break unit tests

Our [unittests break ](https://cirrus-ci.com/task/4762004492648448) in `test_sdist_install_dirty` and `test_sdist_install_with_1_0_tag_dirty`. Strangely those tests fail only under `test (OS X)` and `coverage (Linux - 3.6)`.

This is the stacktrace.
```python
AssertionError: assert (False or False)
  +  where False = <built-in method endswith of str object at 0x103224300>('.d20220109')
  +    where <built-in method endswith of str object at 0x103224300> = 'ge9df8ac.d20220110'.endswith
  +  and   False = <built-in method endswith of str object at 0x103224440>('.d2022010')
  +    where <built-in method endswith of str object at 0x103224440> = 'ge9df8ac.d2022011'.endswith
```
The stacktrace shows that somehow the new automatic generated version string is the date like `d20220109`. This is quite strange as we use `no-guess-dev` for setuptools-scm's configuration. Strange is that the current setuptools-scm version is `6.3.2` from September 21th, 2021. So why does this appear now?


## Versions and main components

* PyScaffold Version: master
* Python Version: *
* Operating system: Mac/Linux
* How did you install PyScaffold: (conda/pip)

--- a/tests/test_install.py
+++ b/tests/test_install.py
@@ -13,9 +13,9 @@ can execute in parallel and not interfere with each other.
 import os
 import shutil
 from contextlib import contextmanager
+from datetime import datetime, timezone
 from pathlib import Path
 from shutil import copyfile
-from time import strftime
 
 import pytest
 
@@ -183,24 +183,22 @@ class DemoApp:
 
 
 def check_version(version, exp_version, dirty=False):
-    today = int(strftime("%Y%m%d"))
-    yesterday, tomorrow = today - 1, today + 1
-    # ^  sometimes the day in the dirty tag has a 1-off error ¯\_(ツ)_/¯
-    #    specially in the case the test is running around midnight
-    dirty_tags = [f".d{d}" for d in (today, yesterday, tomorrow)]
+    today = datetime.now(timezone.utc).strftime("%Y%m%d")
+    dirty_tag = f".d{today}"
     # ^  this depends on the local strategy configured for setuptools_scm...
     #    the default 'node-and-date'
 
-    # for some setuptools version a directory with + is generated, sometimes _
     sep = "+" if "+" in version else "_"
+    # ^  for some setuptools version a directory with + is generated, sometimes _
+
     if dirty:
         ver, local = version.split(sep)
-        assert any(local.endswith(d) for d in dirty_tags)
+        assert local.endswith(dirty_tag)
         assert ver == exp_version
     else:
         ver = version.split(sep)
         if len(ver) > 1:
-            assert not any(ver[1].endswith(d) for d in dirty_tags)
+            assert not ver[1].endswith(dirty_tag)
         assert ver[0] == exp_version
 
 

--- a/tests/test_install.py
+++ b/tests/test_install.py
@@ -183,26 +183,24 @@ class DemoApp:
 
 
 def check_version(version, exp_version, dirty=False):
-    dirty_tag = ".d" + strftime("%Y%m%d")
+    today = int(strftime("%Y%m%d"))
+    yesterday, tomorrow = today - 1, today + 1
+    # ^  sometimes the day in the dirty tag has a 1-off error ¯\_(ツ)_/¯
+    #    specially in the case the test is running around midnight
+    dirty_tags = [f".d{d}" for d in (today, yesterday, tomorrow)]
     # ^  this depends on the local strategy configured for setuptools_scm...
     #    the default 'node-and-date'
 
     # for some setuptools version a directory with + is generated, sometimes _
+    sep = "+" if "+" in version else "_"
     if dirty:
-        if "+" in version:
-            ver, local = version.split("+")
-        else:
-            ver, local = version.split("_")
-        assert local.endswith(dirty_tag) or local[:-1].endswith(dirty_tag[:-1])
-        # ^  sometimes the day in the dirty tag has a 1-off error ¯\_(ツ)_/¯
+        ver, local = version.split(sep)
+        assert any(local.endswith(d) for d in dirty_tags)
         assert ver == exp_version
     else:
-        if "+" in version:
-            ver = version.split("+")
-        else:
-            ver = version.split("_")
+        ver = version.split(sep)
         if len(ver) > 1:
-            assert not ver[1].endswith(dirty_tag)
+            assert not any(ver[1].endswith(d) for d in dirty_tags)
         assert ver[0] == exp_version
 
 

